File: app/src/main/java/com/donygeorge/todolist/EditItemActivity.java
package com.donygeorge.todolist;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.EditText;

public class EditItemActivity extends AppCompatActivity {

    private int mIndex;
    private EditText mItemEditText;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_edit_item);
        String text = getIntent().getStringExtra("text");
        mIndex = getIntent().getIntExtra("index", -1);
        mItemEditText = (EditText) findViewById(R.id.itemEditText);
        mItemEditText.setText(text);
        mItemEditText.setSelection(mItemEditText.getText().length());
        mItemEditText.requestFocus();
    }

    public void onSave(View v) {
        // Prepare data intent
        Intent data = new Intent();
        data.putExtra("text", mItemEditText.getText().toString());
        data.putExtra("index", mIndex);
        setResult(RESULT_OK, data);
        finish();
    }
}

File: app/src/main/java/com/donygeorge/todolist/EditItemActivity.kt
package com.donygeorge.todolist

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.EditText

class EditItemActivity : AppCompatActivity() {

    private var mIndex: Int = 0
    private var mItemEditText: EditText? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_edit_item)
        val text = intent.getStringExtra("text")
        mIndex = intent.getIntExtra("index", -1)
        mItemEditText = findViewById(R.id.itemEditText) as EditText
        mItemEditText!!.setText(text)
        mItemEditText!!.setSelection(mItemEditText!!.text.length)
        mItemEditText!!.requestFocus()
    }

    fun onSave(v: View) {
        // Prepare data intent
        val data = Intent()
        data.putExtra("text", mItemEditText!!.text.toString())
        data.putExtra("index", mIndex)
        setResult(Activity.RESULT_OK, data)
        finish()
    }
}
